### DISCUSSION TITLE
Column width or newline characters

### DISCUSSION THREAD

**Discussion by paalaas** [fbaligand/kibana-enhanced-table#299] [Q&A] [answered]
I'm using the Document table visualization, and one of the columns has quite an amount of text, which I would like to limit the column width for. Is this possible with the plug-in either by setting the width in some way or by adding newline characters in the text itself?

**Comment by fbaligand** [comment, marked as answer]
Hi, you can do that:
- add a computed column
- Set formula: col['fieldName']
- use « cell computed css» to custom width or any other cell CSS property.
- Set “custom position” to put the computed column wherever you want. 
- finally, in hidden columns, set : `fieldName`  
it allows to mask original field column.

**Reply by paalaas** [reply]
OK thx I was actually on that path :-), is there support for newline characters in the text shown for word wrapping purpose?

**Reply by fbaligand** [reply]
I’m not sure to understand. 
what do you mean concretely?

**Reply by paalaas** [reply]
If I can insert newline character (\n \r) in the text shown in a column that will cause a line feed ie that the text break to a new line?

**Reply by fbaligand** [reply]
Have you tried “width” css property?

or else try this:
`word-wrap: break-word; overflow-wrap: break-word; `

**Reply by fbaligand** [reply]
And globally, as CSS is a big web standard, you can search on google how to do this. 
That’s what I did to answer you.

**Reply by paalaas** [reply]
OK yes have tried "width" and it works to some extent, but I was wondering if I could insert a HTML or other newline character directly in text  like "cat\<br>mouse"

**Reply by fbaligand** [reply]
In enhanced table computed column, you can enable "Apply template", and in "Template" setting, you can insert whatever HTML you want.
For instance:
`<div class="some-class" style="some-style">{{value}}</div>`

Then, if in computed column, there are some special text that you want to replace by `<br>`, you can use `replace` function in Formula:
`replace(col1, '<some-special-sequence>', '<br>')`

**Reply by paalaas** [reply]
I have tested it now and and I see the text in the column as below and has applied template with \<div class="some-class" style="some-style">{{value}}\</div> but it doesn't seem to react on \<br> in HTML inserted

"mouse\<br>cheese
cat\<br>mouse"

**Reply by paalaas** [reply]
Found it needed to be an addtional set of curly brackets to avoid HTML escaping {{{value}}}

**Reply by fbaligand** [reply]
Indeed! By default, HTML special chars like `<>` in `value` are escaped, unless you use `{{{value}}}`.
This is a [mustache](https://mustache.github.io/mustache.5.html) templating feature.
